packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -895,6 +1080,687 @@ void Add(
   }
 }
 
+//Given array containing entries per row, produce rowptrs
+//Used by all add kernels
+template<typename RowPtrType>
+struct PrefixSumFunctor
```

Comment:
Couldn't we just use the existing function `Tpetra::Details::computeOffsetsFromCounts`?